sched/scheduler/stateful_scheduler.go

```diff
@@ -238,11 +243,17 @@ func (s *statefulScheduler) scheduleTasks() {
 		s.asyncRunner.RunAsync(
 			func() error {
 				log.Println("Starting task", taskId, " command:", strings.Join(taskDef.Argv, " "))
-				return runTaskAndLog(
+				status, err := runTaskAndLog(
 					saga,
 					wf,
 					taskId,
 					taskDef)
+				if err == nil && status.State == runner.COMPLETE {
+					s.stat.Counter("schedCompletedTaskCounter").Inc(1)
```

Comment:
I think it's cleaner to have stats grouped together when possible rather than pass StatsReceiver everywhere. And it seems more reasonable to return a result from RunAndLog than not (I'm not a fan of void functions). That said, Dan and I were discussing moving more stats into Runner and further down the stack generally so we may revisit this.